src/models/aspera-sdk.model.ts

```diff
@@ -119,13 +121,46 @@ export class ActivityTracking {
       return;
     }
 
+    this.lastWebSocketEvent = webSocketEvent;
+
+    this.notifyWebSocketEvent(webSocketEvent);
+  }
+
+  /**
+   * Notify all consumers when a connection webSocketEvent occurs.
+   *
+   * @param webSocketEvent the event type.
+   */
+  private notifyWebSocketEvent(webSocketEvent: WebsocketEvent): void {
+    if (this.lastNotifiedWebSocketEvent === webSocketEvent) {
+      return;
+    }
+
     this.event_callbacks.forEach(callback => {
       if (typeof callback === 'function') {
         callback(webSocketEvent);
       }
     });
 
-    this.lastWebSocketEvent = webSocketEvent;
+    this.lastNotifiedWebSocketEvent = webSocketEvent;
+  }
+
+  /**
+   * Notify all consumers when the client changes status. For example, when
+   * IBM Aspera App is launched or closed.
+   *
+   * @param running whether the client is running or not.
+   */
+  handleClientEvents(running: Boolean): void {
+    let webSocketEvent: WebsocketEvent;
+
+    if (!running) {
+      webSocketEvent = 'CLOSED';
+    } else {
+      webSocketEvent = this.lastWebSocketEvent;
```

Comment:
I would expect to see this as `RECONNECT` or whatever the other variant is for `WebsocketEvent`. Is there a logic reason why we assign from the member variable?

IMO would be better if `this.lastWebSocketEvent` can just be encapsulated (i.e. only read from and written to) in `this.notifyWebsocketEvent()`.

Reply:
I think we should handle this separately since the websocket state would not necessarily be `RECONNECT` when the client is running, that's why I return `this.lastWebSocketEvent`